tests/ErrorTest.hack

```diff
@@ -244,6 +244,38 @@ final class ErrorTest extends PlaygroundTest {
                     ],
                 ),
             ),
+
+            'invalid enum values' => tuple(
+                'query { takes_favorite_color(favorite_color: foo) }',
+                dict[],
+                shape(
+                    'data' => dict[
+                        'takes_favorite_color' => null
+                    ],
+                    'errors' => vec[
+                        shape(
+                            'message' => 'Expected a valid value for FavoriteColor, got Graphpinator\Parser\Value\EnumLiteral',
+                            'path' => vec['takes_favorite_color']
+                        )
+                    ]
+                )
+            ),
+
+            'invalid enum values with arguments' => tuple(
+                'query { takes_favorite_color(favorite_color: $favorite_color) }',
+                dict['favorite_color' => 'foo'],
+                shape(
+                    'data' => dict[
+                        'takes_favorite_color' => null
+                    ],
+                    'errors' => vec[
+                        shape(
+                            'message' => 'Caught exception while resolving field.',
```

Comment:
This error is kind of lame — would it make more sense to call `coerceValue()` instead of `assertValidVariableValue()` in `InputType::coerceNode()`, @jjergus ? The methods do nearly the same thing in all cases, except `coerceValue()` is a little more permissive and has better error messages.

Reply:
Indeed, if this error shows up anywhere besides its intended use (i.e. when the actual Hack method that is annotated with `<<GraphQL\Field>>` throws an exception), that is a bug.

In this case, the bug is that we have not yet implemented [this TODO](https://github.com/mwildehahn/hack-graphql/blob/main/src/Resolver.hack#L68) and query validation. After that is implemented, `assertValidVariableValue()` should be guaranteed to never throw (there's also a comment about it [here](https://github.com/mwildehahn/hack-graphql/blob/main/src/Types/Input/InputType.hack#L41)).

Reply:
Ah, makes sense - I'll leave this as-is for now, then.